### PR TITLE
fixes #116 and jumps to page if provided via argument.

### LINKED ISSUE #116: page parameter/message is silently ignored
The application usage help says thus:

> `Usage: /boot/system/apps/BePDF/BePDF [options] [<PDF-file> [<page>]]`

The source even contains a definition of a page num message in https://github.com/HaikuArchives/BePDF/blob/master/bepdf/beos/BepdfApplication.cpp#L63 and it is used when parsing arguments in
https://github.com/HaikuArchives/BePDF/blob/master/bepdf/beos/BepdfApplication.cpp#L664
and added to the refs received message in
https://github.com/HaikuArchives/BePDF/blob/master/bepdf/beos/BepdfApplication.cpp#L676
but in the actual `RefsReceived` method it is silently ignored, making the parameter useless:
https://github.com/HaikuArchives/BePDF/blob/master/bepdf/beos/BepdfApplication.cpp#L531

This would really be important for scripting, I took a look but the application does not separate between UI and backend code, making this fix harder than necessary...

You can easily test the message handling part via
> `hey BePDF '_RRC' with "bepdf:page_num"=11`

or use the application signature `application/x-vnd.mp-xpdf` to go sure.

### PR DESCRIPTION
**Pull request by grexe**
first attempt, todo:
* test with open documents and jump to page
* check why alert does not work on page parsing error
* translation of said alert

### CODE REVIEW

File: bepdf/beos/BepdfApplication.cpp [hunk 1]
```diff
@@ -545,12 +545,21 @@ void BepdfApplication::RefsReceived ( BMessage * msg )
...
 	if (B_OK == msg->FindString("ownerPassword", &ownerPassword)) {
 		owner = ownerPassword.String();
 	}
 	if (B_OK == msg->FindString("userPassword", &userPassword)) {
 		user = userPassword.String();
 	}
+    status_t result = msg->FindInt32(PAGE_NUM_MSG, &pageNum);
+    if (result != B_OK) {
+        if (result != B_NAME_NOT_FOUND) {
+            BAlert *error = new BAlert(B_TRANSLATE("Error"), B_TRANSLATE("BePDF: Error getting page number!"), B_TRANSLATE("Close"), NULL, NULL, B_WIDTH_AS_USUAL, B_WARNING_ALERT);
```
Comment (humdingerb) [inline bepdf/beos/BepdfApplication.cpp:559]: Alerts normally don't start with the app's name, here "BePDF:".
The starndard button label to acknowledge an alert is "OK".

**Review by humdingerb — COMMENTED** [review]
I tried what suggested at #116.

Neither the system BePDF, nor the one compiled with this patch react on the message `hey BePDF '_RRC' with "bepdf:page_num"=11`

I tried while displaying a PDF with <11 pages  and >11 pages.

**Comment by grexe** [comment]
I just aligned with existing BePDF, should I fix this, too? Would need a change to translations, as the "BePDF:" prefix is used in the message.

**Comment by grexe** [comment]
I spent some quality time with hey and even compiled a debug build to find out what is going on, turns out that :
* hey can actually send a refs_received message via `LOAD` command (same as using `_RRC` for the `B_REFS_RECEIVED` message what code, and 
* the file is specified via the `file` parameter and path value

this worked fine here, even jumping to different pages in the already opened document, just like I wanted :*)

```
hey application/x-vnd.mp-xpdf LOAD file\("/boot/home/Desktop/PDF use case/BeOS.scripting.pdf"\) with "bepdf:page_num"=int32\(2\)
```

I also added sanity checking for the page number and limit it to the document boundaries `1 < page_number < count_pages`.

**Comment by humdingerb** [comment]
OK, I may have misunderstood the `hey BePDF '_RRC' with "bepdf:page_num"=11` mention in #116 . Thought if you had a PDF open, that line was supposed to go to page 11...

Anyway, it does work fine, also when using the command line options.

If the "BePDF:" prefix and "Close" is used in alerts throughout BePDF, we dont have to start changing it here...

Let's merge! Thanks for the PR!

**Pull request merged**